### PR TITLE
Devise Invitations & Plan CRUD

### PR DESCRIPTION
**Pull request by abrahamzaragoza**
- Add Devise Invitatations - invite user through mail
- Update Navbar
- Add Money gem
- Add Plan Model & CRUD
- Add Rubucop-Rails & Rubocop-Rspec gems
- Add Rspec gem & define rails generator test framework
- Add Plan specs
- Add Plan seed

### CODE REVIEW

File: app/controllers/plans_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class PlansController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:index]
```
Comment (ohenroute) [inline app/controllers/plans_controller.rb]: why do we do this?

File: app/controllers/plans_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @plan = Plan.new
+  end
+
+  def edit; end
+
+  def create
+    if can_create_plan?
+      @plan = Plan.new(plan_params)
+      if @plan.save
+        raise_flash_and_redirect_to(:notice, 'Plan was created successfully.', plans_path)
```
Comment (ohenroute) [inline app/controllers/plans_controller.rb]: great job using modules

File: app/controllers/plans_controller.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      else
+        render 'new'
+      end
+    else
+      raise_flash_and_redirect_to(:alert, 'The maximum amount of plans have been reached.', plans_path)
```
Comment (ohenroute) [inline app/controllers/plans_controller.rb]: why raise? are you raising an error, though?

File: test/channels/application_cable/connection_test.rb [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
-  #   cookies.signed[:user_id] = 42
-  #
-  #   connect
-  #
-  #   assert_equal connection.user_id, "42"
-  # end
+require 'test_helper'
+
+module ApplicationCable
+  class ConnectionTest < ActionCable::Connection::TestCase
```
Comment (ohenroute) [inline test/channels/application_cable/connection_test.rb:6]: it's not a good practice to merge commented code.

**Review by ohenroute — CHANGES_REQUESTED** [review]
Try not adding so much overhead. 

There are many files created. Are we really need to use them all? Does the cost of the gem used is enough as to justify it?

File: app/controllers/plans_controller.rb (hunk 1 above)
Comment (abrahamzaragoza) [reply app/controllers/plans_controller.rb]: This one is lo longer needed, all current pages related to Plans are for admins.

File: app/controllers/plans_controller.rb (hunk 3 above)
Comment (abrahamzaragoza) [reply app/controllers/plans_controller.rb]: Not really, removed the word raise from the method.

File: app/views/plans/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<section class='container-fuild col-8 col-sm-6 offset-sm-3 text-bg-purple p-3 rounded-3'>
+  <%= form_with model: @plan do |f|  %>
```
Comment (ohenroute) [inline app/views/plans/_form.html.erb:2]: please also research form_for

File: test/controllers/plans_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class PlansControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (ohenroute) [inline test/controllers/plans_controller_test.rb]: commented code

File: test/models/user_test.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (ohenroute) [inline test/models/user_test.rb]: commented code.

**Review by ohenroute — CHANGES_REQUESTED** [review]

File: app/views/plans/_form.html.erb (hunk 5 above)
Comment (abrahamzaragoza) [reply app/views/plans/_form.html.erb:2]: Isn't it soft-deprecated? [Ruby Guides](https://guides.rubyonrails.org/form_helpers.html#using-form-tag-and-form-for)

File: app/controllers/plans_controller.rb (hunk 1 above)
Comment (ohenroute) [reply app/controllers/plans_controller.rb]: understood, will we have another authenticate_user?

File: app/views/plans/_form.html.erb (hunk 5 above)
Comment (ohenroute) [reply app/views/plans/_form.html.erb:2]: that's correct. but you didn't change the devise forms. please keep them consistent.

**Review by ohenroute — APPROVED** [review]
Good job 👍

**Pull request merged**